lib/install/management/create.go

```diff
@@ -100,10 +102,69 @@ func (d *Dispatcher) createPool(conf *config.VirtualContainerHostConfigSpec, set
 	return nil
 }
 
-func (d *Dispatcher) startAppliance(conf *config.VirtualContainerHostConfigSpec) error {
+// relocateAppliance invokes the host placement library to find a suitable host (based on host
+// metrics) to relocate (migrate) the VCH to. If the VCH's current host is suitable, the VCH
+// is not relocated.
+func (d *Dispatcher) relocateAppliance() error {
 	defer trace.End(trace.Begin("", d.op))
+	var err error
+
+	// TODO(anchal): Use ranked host policy when #7459 is merged.
+	// provider := performance.NewHostMetricsProvider(d.session)
+	// rankedPolicy := placement.NewRankedHostPolicy()
+
+	randomPolicy := placement.NewRandomHostPolicy()
+	if randomPolicy.CheckHost(d.op, d.appliance) {
+		// The current host of the appliance is suitable; no migration needed.
+		return nil
+	}
+
+	host, err := randomPolicy.RecommendHost(d.op, d.appliance)
+	if err != nil {
+		msg := "Unable to obtain recommended host: %s"
+		d.op.Warnf(msg, err)
+		return errors.Errorf(msg, err)
+	}
+
+	hMoref := host.Reference()
+	d.op.Debugf("Attempting to relocate VCH to host %s", hMoref.String())
+	relocateSpec := types.VirtualMachineRelocateSpec{
+		Host: &hMoref,
+	}
+
+	_, err = d.appliance.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
+		return d.appliance.Relocate(ctx, relocateSpec, types.VirtualMachineMovePriorityDefaultPriority)
+	})
+	if err != nil {
+		msg := "Unable to relocate VCH: %s"
+		d.op.Warnf(msg, err)
+		return errors.Errorf(msg, err)
+	}
+
+	// TODO(anchal): Undo this step if the folder path is unaffected.
+	if d.vmPathName, err = d.appliance.FolderName(d.op); err != nil {
+		d.op.Errorf("Failed to get canonical name for appliance: %s", err)
+		return errors.New(err.Error())
+	}
+
+	return nil
+}
 
+func (d *Dispatcher) startAppliance(conf *config.VirtualContainerHostConfigSpec) error {
+	defer trace.End(trace.Begin("", d.op))
 	var err error
+
+	// If DRS is disabled (e.g. in a ROBO env), find a suitable host and relocate the VCH to it.
+
+	// TODO(anchal): Add license check for ROBO once #7283 is closed.
+	// TODO(anchal): Enable this check once #7440 is merged.
+	// if conf.DRSDisabled {
+	err = d.relocateAppliance()
```

Comment:
Basically, I am saying that all logging could be done here.

Reply:
That makes sense. Agreed that having just one error log in the caller is preferred over multiple logs in the function being called. Currently, log statements are placed in `relocateAppliance` so we can capture the exact error (as there are multiple possible error scenarios) and then log the "main" warning `"Unable to relocate VCH, attempting to use its current host for powering on"` separately.

This avoid us having to log a run-on message in the caller like:

`"Unable to relocate VCH due to: Unable to obtain recommended host: <error>"`

or

`"Unable to relocate VCH due to: Failed to get canonical name for appliance:: <error>"`

While the current approach trades number of log statements for more readability, the suggested approach does the opposite. Which should we pick?

(Also this is a WIP changeset and I'll be revising it frequently.)

Reply:
I would not call existing state of logs as easily readable ;)
If you are worried about compressing too much info into one line, you could easily log it as two lines:
```
log.op.Warn(err)
log.op.Warn("Attempting to use its current host for powering on")
```

Reply:
Perfect, fixed. Thanks!